src/mfs.jl

```diff
@@ -0,0 +1,163 @@
+    If the norm is smaller, it updates the best pertubation found so far.
+    It repeats this process Ntries times and returns the best pertubation found.
+
+"""
+
+function random_pertubation(mapper::AttractorMapper, X, search_area, dim,  Ntries=10000)
+    best_dist = Inf
+    best_shock = nothing
+    Xsp = StateSpaceSet([X])
+    for _ in 1:Ntries
+        u0 = search_area[1] .+ rand(dim) .* (search_area[2] .- search_area[1])
+        shock = X + u0
+        if !(check_basin(mapper, shock, X))
+            dist = LinearAlgebra.norm(u0)
+            if dist < best_dist
+                best_dist = dist
+                best_shock = u0
+            end
+        end
+    end
+    return best_shock, best_dist
+end
+
+
+""" 
+    mfs_brute_force(mapper, X, best_shock, best_dist, dim,  Ntries=10000)
+
+Input:
+    mapper: AttractorMapper
+    X: initial point to test
+    best_shock: vector of the best shock found so far
+    best_dist: norm of the best shock found so far
+    dim: dimension of the system
+    Ntries: number of random pertubations to try, default = 10000
+
+Output:
+    best_shock: vector of the best shock found
+    best_dist: norm of the best shock found
+
+Description:
+    This function works on the results obtained by random_pertubation(). It starts from the best shock found so far and tries to find a better one 
+    by continuously reducing the radius of the sphere on the surface of which it generates random pertubations. If pertubation with the same basin of attraction is found,
+    it updates the best shock found so far and reduces the radius of the sphere. 
+    It repeats this process Ntries times and returns the best pertubation found.
+"""
+
+
+function mfs_brute_force(mapper::AttractorMapper, X, best_shock, best_dist, dim,  Ntries=10000)
+    
+    radius_dif = 100.
+
+    for _ in 1:Ntries
+
+        pertubation = random_point_on_sphere((best_dist-best_dist/radius_dif), dim)
+        new_shock = X + pertubation
+        
+        if !(check_basin(mapper, X, new_shock))
+            best_dist = set_distance(StateSpaceSet([X]), StateSpaceSet([new_shock]), Centroid())
+            best_shock = pertubation
+        end
+    end
+
+    return best_shock, best_dist
+end
+
+
+"""
+    mfs(mapper, search_area, X, dim, n_iterations = 10000)
+
+Input:
+    mapper: AttractorMapper
+    search_area: array of two points defining the search area
+    X: initial point to test
+    dim: dimension of the system
+    n_iterations: number of iterations to run each step of the algorithm, default = 10000
+
+Output:
+    best_shock: vector of the best shock found
+    best_dist: norm of the best shock found
+
+Description:
+    This function runs the minimal fatal shock algorithm on the initial point X.
+    It does so by first generating a random pertubation of X and then trying to find a better one by using the brute force method.
+"""
+
+
+function mfs(mapper::AttractorMapper, search_area, X, dim, n_iterations = 10000)
```

Comment:
Please rename to `minimal_fatal_shock`. Also make `n_iterations` a keyword, and name it something more intuitive, like `total_iterations`. ("n" means nothing, `n_iterations` is just as descriptive and intuitive as `iterations`).

Reply:
Actually, `n_iterations` should become an argument of the new Type that will specify the Brute Force approach, see below.